### PR TITLE
Eliminate B4B attempt #2

### PR DESCRIPTION
**Pull request by johnmauff**
This commit eliminates a number of code blocks that are designed to produce B4B answers between the CPU and GPU versions of the code. In particular, it focuses on code blocks that contain math intrinsic functions that have previously caused numerical differences. Recently a flag has been added to the NVHPC compiler 'math_uniform' that provides a math intrinsic function calculations on the GPU that reproduces the CPU version.

### CODE REVIEW

File: src/misclibs.F [hunk 1]
```diff
@@ -3678,9 +3670,11 @@ subroutine totmois(nstat,rstat,train,ruh,rvh,rmh,qv,ql,qi,rho)
       !$acc end parallel
 #endif
  
+#ifdef _B4B07R
```
Comment (sjsprecious) [inline src/misclibs.F:3673]: Do we have to keep this BFB flag?

File: src/sfcphys.F [hunk 2]
```diff
@@ -1216,15 +1152,15 @@ subroutine getavgsfc(u1,v1,s1,t1,avgsfcu,avgsfcv,avgsfcs,avgsfct)
       call MPI_ALLREDUCE(MPI_IN_PLACE,avgsfct,1,MPI_DOUBLE_PRECISION,MPI_SUM,MPI_COMM_WORLD,ierr)
 #endif
 
-#ifndef _B4B10
+#ifndef _B4B10R
```
Comment (sjsprecious) [inline src/sfcphys.F:1155]: Do we need to keep this BFB flag?

File: src/sounde.F [hunk 3]
```diff
@@ -75,15 +75,19 @@ subroutine sounde(dt,xh,arh1,arh2,uh,ruh,xf,uf,yh,vh,rvh,yf,vf,     &
...
     IF( nrkmax.eq.3 )THEN
       if(nrk.eq.1)then
 !!!        nloop=1
 !!!        dts=dt/3.
         nloop=nint(float(nsound)/3.0)
-        dts=dt/(nloop*3.0)
+!        print *,'sounde: denominator: ', float(nloop)*3.0
+!        print *,'sounde: numerator: ', dt
+        dts=dt/(float(nloop)*3.0)
+!        print *,'sounde: dts: ', dts
```
Comment (sjsprecious) [inline src/sounde.F]: can we delete these print statements?

File: src/sounde.F [hunk 4]
```diff
@@ -75,15 +75,19 @@ subroutine sounde(dt,xh,arh1,arh2,uh,ruh,xf,uf,yh,vh,rvh,yf,vf,     &
 
 !---------------------------------------------------------------------
 
+    print *,'sounde: nsound: ',nsound
     IF( nrkmax.eq.3 )THEN
       if(nrk.eq.1)then
 !!!        nloop=1
 !!!        dts=dt/3.
```
Comment (sjsprecious) [inline src/sounde.F:81]: maybe just remove these two lines?

File: src/turb.F [hunk 5]
```diff
@@ -671,10 +659,7 @@ subroutine sfc_and_turb(getsfc,getpbl,nstep,dt,dosfcflx,cloudvar,qbudget,    &
               ustt(i,j) = max( st(i,j)*karman/alog((zf(i,j,nk+1)-zh(i,j,nk))/cnst_znt) , 1.0e-6 )
             enddo
             enddo
-#ifdef _B4B01F
-            !$acc update device(ut,vt,st,ustt)
-#endif
-            stop 'sfc_and_turb: after setting of ut,vt,st,ustt'
+            !stop 'sfc_and_turb: after setting of ut,vt,st,ustt'
```
Comment (sjsprecious) [inline src/turb.F]: can we remove this line?

File: src/misclibs.F (hunk 1 above)
Comment (johnmauff) [reply src/misclibs.F:3673]: @sjsprecious, Note that I am in the process of removing all of the B4B section of code. I am not yet done.

File: src/sfcphys.F (hunk 2 above)
Comment (johnmauff) [reply src/sfcphys.F:1155]: Still in the process of removing all B4B sections of the code.  I am not yet done.

File: src/sounde.F (hunk 4 above)
Comment (johnmauff) [reply src/sounde.F:81]: This code was added to CM1 by another author.  No idea what it is still present

File: src/turb.F (hunk 5 above)
Comment (johnmauff) [reply src/turb.F]: Yes I will delete this line

File: src/sounde.F (hunk 3 above)
Comment (johnmauff) [reply src/sounde.F]: Yes, I will delete this line.

**Review by sherimickelson — APPROVED** [review]

**Review by sjsprecious — APPROVED** [review]

**Pull request merged**